### ISSUE TITLE
Travis CI announces sbt 1.2.6 but uses 0.13.18 

### ISSUE THREAD

**Issue by LeeTibbert** [scala-native/scala-native#1594]
#### tl;dr
Travis CI announces it is using SBT_VERSION 1.2.6 for link-order and probably
other tests, especially scripted tests.

From yesterday's Travis log.  Top:
```
JDK: oraclejdk8 Scala: 2.11.8
MODE=test-runtime TARGET_DOCKER_PLATFORM=library SCALANATIVE_MODE=release-fast SCALANATIVE_GC=immix SBT_VERSION=1.2.6 TEST_COMMAND="sbtScalaNative/scripted"
```
But it is actually using sbt 0.13.18, link-order section:
```
Running run / link-order
[info] Getting org.scala-sbt sbt 0.13.18  (this may take some time)...
[info] :: retrieving :: org.scala-sbt#boot-app
[info] 	confs: [default]
[info] 	49 artifacts copied, 0 already retrieved (17584kB/103ms)
[info] Getting Scala 2.10.7 (for sbt)...
[info] :: retrieving :: org.scala-sbt#boot-scala
[info] 	confs: [default]
[info] 	5 artifacts copied, 0 already retrieved (24560kB/51ms)
[info] [info] Updated file /tmp/sbt_7636871/link-order/project/build.properties: set sbt.version to 0.13.18
```
I believe that part of the problem comes from file ci-docker/Dockerfile 
having the line:`ENV SBT_LAUNCHER_VERSION 0.13.7 `.

I suspect, but do not know, that the 0.13.7 ought to be the $SBT_VERSION used
in the last line of the Dockerfile. 

#### Background

For a number of months now I have been experience a failure with the link-order
test when I run test-all with sbt 1.2.n.  Each time I read a log file I have to quiet
my nerves and remember that error as a known error.  Not the greatest 
aggravation in my life but it disturbs my tranquility.

I submitted PR #1446 which contained the necessary change for link-order to
work with both 0.13.x and 1.2.x.  It has not been excepted because I could
not adequately explain why I was seeing the error and Travis was not.

I believe that I now have that explanation.  

I think the first step is to correct the Dockerfile to use the 1.2.n sbt-runner when
executing in the 1.2.6 sbt environment.  I believe that my link-order failure will
then become evident.

I did not dive deeper into the levels of sbt version. The SBT_VERSION on
the last line in the file appears OK>

Now, the keen mind will ask why would one invite failure when Travis is working
with 1.2.6 right now.  I think the 1.2.6 environment is not using the intended
SN plugin and/or SBT library environment.

**Comment by ekrich** [comment]
I would like to point out that the sbt version on master is now `1.2.8` in build.sbt. From what I can see on the command line you should invoke as `sbt ^^1.2.8 rebuild test-all` or whatever commands are being called which looks correct. You should only see `2.11` and `2.12` output for tools and such published locally - no `2.10` when using ``1.2.8`. The `0.13.x` version should just use the one found in `project/build.properties` - `0.13.18` in this case. I guess I should have changed the versions in Travis to match the ones in build.sbt. The following should also be changed in the `ci-docker/Dockerfile` - `ENV SBT_LAUNCHER_VERSION 0.13.7` as we should launch on the same version as we are going to use.

**Comment by shadaj** [comment]
Need to take a closer look at this tomorrow, but `SBT_LAUNCHER_VERSION` refers to the version of SBT that is installed when building the container image, not what is used at runtime. That version is selected by an environment variable specified in the Travis matrix.

It would probably help to use `1.2.8` as the `SBT_LAUNCHER_VERSION`, since then that avoids downloading launcher jars that are unused.

**Comment by LeeTibbert** [comment]
@shadaj  Thank you for looking at this issue.  And they call Computer Science the study of 
complexity...  Plenty of that here.

**Comment by ekrich** [comment]
@LeeTibbert Do you think this could be closed now?

**Comment by LeeTibbert** [comment]
Sorry, I thought that I had replied earlier that I could follow your judgement, since you untangled this in the PR which brought SN up to 1.3.7 (thank you for fixing all that).

When SN PR acceptance ramps up I'll be examining Travis CI logs closely again.  If I see anything
strange on this topic, I will open a new Issue.

Thanks for the gardening of the Issues list.
